apps/backend/src/datasources/postgres/EventLogsDataSource.ts

```diff
@@ -23,15 +23,25 @@ export default class PostgresEventLogsDataSource
     eventType: string,
     rowData: string,
     changedObjectId: string,
-    description?: string
+    description?: string,
+    impersonatingUserId?: number
   ) {
+    const isImpersonating =
+      impersonatingUserId != null && impersonatingUserId !== changedBy;
+
+    const actingUserId = impersonatingUserId ?? changedBy;
+
+    const updatedDescription = isImpersonating
+      ? `${description || ''} (${impersonatingUserId} impersonating ${changedBy})`
+      : description;
+
     return database
       .insert({
-        changed_by: changedBy,
+        changed_by: actingUserId,
         event_type: eventType,
         row_data: rowData,
         changed_object_id: changedObjectId,
-        description: description,
+        description: updatedDescription,
```

Comment:
suggestion: Do you think we can store the `impersonatingUserId` as a column in the `event_logs`? This would be very nice and would allow us to associate the actual user object. We can then query and display this information more flexible. I see you already have the value in the method, it would be just adding a new db_patch with column and setting it.